### PR TITLE
WIP: error handling

### PR DESCRIPTION
**Pull request by tathanhdinh**
The PR does:
 - Error handling #2 
 - Rename `Xc` #1

### CODE REVIEW

**Comment by tathanhdinh** [comment]
It would not compile since it depends on a [PR](https://github.com/Wenzel/xenctrl-sys/pull/2) of [`xen-sys`](https://github.com/Wenzel/xenctrl-sys)

File: src/lib.rs [hunk 1]
```diff
@@ -1,90 +1,116 @@
...
+    ptr::{null_mut, NonNull},
+};
+
+use xenctrl_sys::{
+    xc_clear_last_error, xc_domain_maximum_gpfn, xc_domain_pause, xc_domain_unpause, xc_error_code,
+    xc_get_last_error, xc_interface, xc_interface_close, xc_interface_open, xc_monitor_disable,
+    xc_monitor_enable, xentoollog_logger,
+};
 
-pub const PAGE_SHIFT: u32 = xenctrl_sys::XC_PAGE_SHIFT;
```
Comment (Wenzel) [inline src/lib.rs:6]: You need to export these constants as well

**Review by Wenzel — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (tathanhdinh) [reply src/lib.rs:6]: Sorry, I forgot this. PR updated.

File: src/lib.rs [hunk 2]
```diff
@@ -1,90 +1,117 @@
...
+pub struct XenControl {
+    handle: NonNull<xc_interface>,
+    evtchn_port: u32,
 }
 
-impl Xc {
-
-    pub fn new() -> Result<Self,Error> {
+impl XenControl {
+    pub fn from(
```
Comment (Wenzel) [inline src/lib.rs]: Why did you replaced the `new` function, which is a convention to have an object constructor, by the name `from` ?
You break the API here, and i don't understand what this brings us ?

**Review by Wenzel — CHANGES_REQUESTED** [review]

**Comment by tathanhdinh** [comment]
Sorry, I will change this.

**Comment by Wenzel** [comment]
Merging this one !

**Pull request merged**